ntirpc/rpc/svc.h

```diff
@@ -526,10 +526,10 @@ static inline void svc_destroy_it(SVCXPRT *xprt,
 		__warnx(TIRPC_DEBUG_FLAG_SVC_VC,
 			"%s: Connection already closed, hence fd %d closed",
 			 __func__, xprt->xp_fd);
-		xprt->xp_fd = RPC_ANYFD;
```

Comment:
I think that, since we called close() on the FD, we have to clear it.  It's not valid anymore.  I'm not sure I understand how a backchannel can use a closed FD?

Reply:
There's a race between the connection termination and the backchannel. So it's either this solution that will cause socket operations to fail, or replacing such asserts with error codes: https://github.com/nfs-ganesha/ntirpc/blob/eb1937029b2d7425bd8517dd2f5bab2f3cdd9a03/src/svc_vc.c#L368
Since the socket operations can fail anyway, and we already have error handling paths for them, I preferred to use existing paths. But adding new error paths is legit too.
Please let me know.

Reply:
I think that if the client closed the fore-channel, then the back-channel should fail to create.  So probably that assert needs to be a check and fail instead.  Note that clnt_vc_ncreate_svc() is called from Ganesha, so it shouldn't have to deal with internals of ntircp, so failure is probably the best option.

Frank, do you have an opinion on this?

Reply:
Alternatively, we can use [shutdown](https://man7.org/linux/man-pages/man2/shutdown.2.html) instead of close at this point in the code. This can make the connection termination more thread safe, since the file descriptor will be valid as long as there are references to XPRT.

Reply:
We discovered another type of race here: when calling close() the file descriptor is released, and it can be reused by a newer XPRT. But there are still references to the old XPRT, and operations done on the older XPRT might affect the newer XPRT. Thus, changing the assert to an error path wouldn't be enough in this case.

The suggested solution is to call shutdown() instead of close(). Only after we're sure there are no references left to the XPRT we call close().
Pushed a revised version for your review.